pkg/cmd/pulumi/schema_generate_sdk.go

```diff
@@ -0,0 +1,164 @@
+	"path"
+	"path/filepath"
+	"sort"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/spf13/cobra"
+
+	dotnetgen "github.com/pulumi/pulumi/pkg/v3/codegen/dotnet"
+	gogen "github.com/pulumi/pulumi/pkg/v3/codegen/go"
+	nodejsgen "github.com/pulumi/pulumi/pkg/v3/codegen/nodejs"
+	pythongen "github.com/pulumi/pulumi/pkg/v3/codegen/python"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+)
+
+func newSchemaGenerateSDKCommand() *cobra.Command {
+	var target string
+	var outputDirectory string
+
+	cmd := &cobra.Command{
+		Use:   "generate-sdk",
+		Args:  cmdutil.ExactArgs(1),
+		Short: "Generate language SDKs from a Pulumi package schema",
+		Long: "Generate language SDKs from a Pulumi package schema.\n" +
+			"\n" +
+			"Given a Pulumi package schema, write out language SDKs for the package.\n" +
+			"The schema is validated as if by `pulumi schema check` prior to generation.\n" +
+			"\n" +
+			"Valid targets are dotnet, go, nodejs, and python. The 'all' target will\n" +
+			"generate an SDK for each suported target.",
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			if outputDirectory == "" {
+				return fmt.Errorf("an output directory must be specified")
+			}
+
+			pkgSpec, err := loadSchemaSpec(args[0], "")
+			if err != nil {
+				return err
+			}
+
+			pkg, diags, err := schema.BindSpec(pkgSpec, nil)
+			if len(diags) != 0 {
+				diagWriter := hcl.NewDiagnosticTextWriter(os.Stderr, nil, 0, true)
+				wrErr := diagWriter.WriteDiagnostics(diags)
+				contract.IgnoreError(wrErr)
+
+				if diags.HasErrors() {
+					return fmt.Errorf("too many errors")
+				}
+			}
+			if err != nil {
+				return err
+			}
+
+			var files map[string][]byte
+			switch target {
+			case "all":
+				files, err = genAllSDKs(pkg)
+			case "dotnet":
+				files, err = genDotnetSDK(pkg)
+			case "go":
+				files, err = genGoSDK(pkg)
+			case "nodejs":
+				files, err = genNodeSDK(pkg)
+			case "python":
+				files, err = genPythonSDK(pkg)
+			default:
+				return fmt.Errorf("unknown target '%v'", target)
+			}
+			if err != nil {
+				return err
+			}
+
+			if err = os.MkdirAll(outputDirectory, 0755); err != nil {
+				return err
+			}
+			paths := make([]string, 0, len(files))
+			for p := range files {
+				paths = append(paths, p)
+			}
+			sort.Strings(paths)
+			for _, path := range paths {
+				realPath := filepath.Join(outputDirectory, filepath.FromSlash(path))
+				if err = os.MkdirAll(filepath.Dir(realPath), 0755); err != nil {
+					return err
+				}
+				if err = os.WriteFile(realPath, files[path], 0600); err != nil {
+					return err
+				}
+			}
+
+			return nil
+		}),
+	}
+
+	cmd.PersistentFlags().StringVarP(&target, "target", "t", "all",
+		"the target language for SDK generation")
+	cmd.PersistentFlags().StringVarP(&outputDirectory, "output-directory", "o", "",
+		"the output directory for the generated SDK")
+
```

Comment:
Perhaps we want to make the `output-directory` flag required?

```suggestion
cmd.MarkPersistentFlagRequired("output-directory")

```